src/cpuid.cpp

```diff
@@ -46,7 +46,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <cstring> // for std::memset
 #endif
 
-#if TORRENT_HAS_ARM
+#if TORRENT_HAS_ARM && TORRENT_HAS_ARM_NEON
```

Comment:
this header wasn't introduced until glibc 2.16 as far as I can tell, so I added that condition. It doesn't seem linux specific, just glibc specific